examples/simple-grep/README.md

```diff
@@ -0,0 +1,129 @@
+There is a basic example usage of Iris and a tutorial-like explanation:
+
+First of all, let's create a newtype wrapper for the main monad of Iris:
```

Comment:
Let's give a link to `grep` and `ripgrep` to give an idea of what we want to build. It would be also beneficial to provide a usage example to immediately say the expected behaviour:

```shell
$ simple-grep --path=README.md --str=contribut
<and the output goes here>
```